### PR TITLE
var.use_name_prefix

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Toggles between `name` and `name_prefix` asg arguments
* Affects `aws_autoscaling_group` and not the `aws_launch_template`

Followed similar

https://github.com/cloudposse/terraform-aws-security-group/blob/c090804ba938edf434b660e771527ce7d94a52f4/main.tf#L4-L5

## why
* To backport existing resources to use this module without recreating resources

## references
* Closes https://github.com/cloudposse/terraform-aws-ec2-autoscale-group/issues/28

### CODE REVIEW

**Review by jamengual — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -425,3 +425,9 @@ variable "custom_alarms" {
   default     = {}
   description = "Map of custom CloudWatch alarms configurations"
 }
+
+variable "use_name_prefix" {
+  type        = bool
+  default     = true
+  description = "If `true`, this will use the asg argument `name_prefix` instead of `name`"
```
Comment (SweetOps) [inline variables.tf:432]: would be nice to add the list of affected  resources to description

File: variables.tf [hunk 2]
```diff
@@ -425,3 +425,9 @@ variable "custom_alarms" {
   default     = {}
   description = "Map of custom CloudWatch alarms configurations"
 }
+
+variable "use_name_prefix" {
+  type        = bool
+  default     = true
```
Comment (SweetOps) [inline variables.tf:431]: should be `false` to kept the current behaviour

**Review by SweetOps — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (nitrocode) [reply variables.tf:432]: done!

File: variables.tf (hunk 2 above)
Comment (nitrocode) [reply variables.tf:431]: Actually the current behavior uses the name prefix...

**Review by osterman — APPROVED** [review]

**Pull request merged**